### ISSUE TITLE
Large CPU overhead when we execute multiple segment fusion groups 

### ISSUE THREAD

**Issue by ftxj** [csarofeen/pytorch#2327]
### 🐛 Describe the bug

I use nsys to profile the below code and find the large gap before launch kernels. Does we have any method to reduce this gap?

In this example, `add`  and `index_select` can't be fused into one kernel (due to src_tv must on global memory), but these operators are in the same fusion group.
```
import torch

def func(inp1, inp2, idx):
    a = torch.add(inp1, inp2)
    return torch.index_select(a, 0, idx)


script = torch.jit.script(func)

in1 = torch.randn(16384, 2, dtype=torch.float, device="cuda")
in2 = torch.randn(16384, 2, dtype=torch.float, device="cuda")
idx1 = torch.randint(0, 16384, (16384,)).to("cuda").to(torch.long)

with torch.jit.fuser("fuser2"):
    for i in range(10):
        script(in1, in2, idx1)
```

## Version:
devel branch

## cmd line:
PYTORCH_NVFUSER_ENABLE="graph_op_fusion" nsys profile -t cuda,nvtx  --stop-on-exit=true -f true -o example.nsys-rep python example.py 

## nsys-rep file
https://drive.google.com/file/d/1qZ647I4Q_fn_BERdFz5YNFr1JNKUWAUO/view?usp=sharing

## Key Gap

![image](https://user-images.githubusercontent.com/17945397/212224097-e13aa91a-8664-46aa-ac38-c3194e402333.png)

## My Hardware
CPU: AMD Ryzen Threadripper 3960X 24-Core Processor
GPU: RTX 3080 Ti

**Comment by kevinstephano** [comment]
The first issue is there isn't any fusion.  The `add` and the `index_select` are executed as separate kernels so there isn't any difference with eager mode and, given the segmentation, CPU overhead was likely added.

Was segmentation expected?

**Comment by ftxj** [comment]
Yes, I know these ops are executed as separated kernels. But I think even if we do not have any fusion, we can get better performance than the eager mode. I hope the overhead between the two kernels is only the kernel launch because we already got the kernel and what we need is only pass parameters, we do not need do dispatching and any other like pytorch eager mode. does

The nvtx say this timeline gap is from the `OutputAlloc` stage and some other initialize step, I think we do not need to actually allocate memory space in this case.  So, such a big gap makes me confused.

**Comment by naoyam** [comment]
> Was segmentation expected?

Yes. Due to the implementation limitation, the input tensor to `index_select` must be a fusion input, so `add + index_select` is not fused.

**Comment by naoyam** [comment]
Is the `OutputAlloc` region just this? 

https://github.com/csarofeen/pytorch/blob/devel/third_party/nvfuser/csrc/executor.cpp#L1022-L1047

If so, the overhead seems to be due to `at::native::empty_strided_cuda`.

**Comment by ftxj** [comment]
> Is the `OutputAlloc` region just this?
> 
> https://github.com/csarofeen/pytorch/blob/devel/third_party/nvfuser/csrc/executor.cpp#L1022-L1047
> 
> If so, the overhead seems to be due to `at::native::empty_strided_cuda`.

Yes. Can this be optimized?
